src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationParsingTests.cs

```diff
@@ -11133,6 +11133,358 @@ public void Class_IncompleteArgumentList_06(bool @struct)
             EOF();
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/40879")]
+        public void ConstructorInitializerWithArrowInsteadOfColon_Base()
+        {
+            UsingTree("""
+                class C
+                {
+                    C() => base() { }
```

Comment:
Let's add some more variations on this. For example, `C() => base(1);`. Or `C() => base(1) => Console.Write(1);`.

Reply:
And same for the `this` tests below. Could we factor these two into a theory?

Reply:
my preference with theories is they not then contain conditional logic within them to specify the expected outcomes (as i find it hard to  mentally then template instantiate each theory out in my head).  I'm good with individual tests that overlap by a lot.